src/main/fc/fc_dispatch.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#include <platform.h>
+
+#include "drivers/system.h"
+#include "fc/fc_dispatch.h"
+
+#define DISPATCH_QUEUE_SIZE     5
+
+typedef struct {
+    dispatchFuncPtr ptr;
+    uint32_t delayedUntil;
+} dispatchItem_t;
+
+static dispatchItem_t queue[DISPATCH_QUEUE_SIZE];
```

Comment:
This implementation makes little sense ..
Either use simple implementation (scan on insert looking for null entry/scan full table on dispatch), or something with good O() efficiency (priority heap sorted by dispatch time).

Another option is linked list, with modules using dispatch being responsible for `dispatchItem_t` allocation (possibly sorted)